### ISSUE TITLE
Language Server support for Lightning Web Components Open Source

### ISSUE THREAD

**Issue by muenzpraeger** [forcedotcom/salesforcedx-vscode#1378]
**Is your feature request related to a problem? Please describe.**
The Language Server that is shipped with the Salesforce Extensions for Visual Studio Code currently only supports Lightning Web Components development within Salesforce DX projects.

With this it's not easy to develop LWCs off platform, as essential functionality is missing (i. e. auto-completion of tags, cmd+click into other components etc.).

**Describe the solution you'd like**
The Language Server should provide capabilities for LWC off platform, which are agnostic to the project type.

**Comment by pogilvie** [comment]
+1. Yes this is needed to build the viability of LWC as a stand-alone framework.

**Comment by ntotten** [comment]
This will happen soon, we are just waiting on legal and security approval.

**Comment by muenzpraeger** [comment]
🥇

**Comment by ChuckJonas** [comment]
can someone please explain why a javascript framework needs its own language server?  I would think it would only apply to on-platform features (like getting SObject/API information?).

Seems like just providing @types would be enough...

**Comment by muenzpraeger** [comment]
Well, the language server is what for example auto-provisions the types.

Or cmd+click on a JS expression within the HTML template to get routed to the JS file. Other frameworks like Vue have the same.

**Comment by ChuckJonas** [comment]
yea I didn't consider the html templating syntax.

**Comment by mshanemc** [comment]
@ChuckJonas there's also some slightly funky things that html doesn't like when it tries to parse that as actual CSS.  It'd be nice for that work in any LWC context.
`<c-whatever style={computedStyle}></c-whatever>`

**Comment by ntotten** [comment]
Duplicate of #1599
